packages/@react-aria/focus/test/FocusScope.test.js

```diff
@@ -526,4 +527,43 @@ describe('FocusScope', function () {
       expect(document.activeElement).toBe(item1);
     });
   });
+  describe('nested focus scopes', function () {
+    it('should make child FocusScopes the active scope regardless of DOM structure', function () {
+      function ChildComponent(props) {
+        return ReactDOM.createPortal(props.children, document.body);
+      }
```

Comment:
Rendered the child FocusScope in a portal to test the case where the child is not contained within the parent FocusScope in the DOM but is still a child of the parent FocusScope (nested Dialog test case basically)